### PR TITLE
"Choosing the Right Helm Resource" update for Chart v4

### PR DESCRIPTION
**Pull request by EronWright**
## Description

_HOLD until Chart v4 is released with pulumi-kubernetes v4.12._

This PR updates the "Choosing the Right Helm Resource For Your Use Case" guide to use the latest and greatest `Chart` v4 resource.

Closes https://github.com/pulumi/pulumi-kubernetes/issues/2993

### CODE REVIEW

File: themes/default/content/registry/packages/kubernetes/how-to-guides/choosing-the-right-helm-resource-for-your-use-case.md [hunk 1]
```diff
@@ -74,12 +75,24 @@ If you have existing Helm Releases deployed through a version of the Helm CLI an
...
+
+`Helm Release` does not have the same flexibility in offering transformations support, aside from applying a post-renderer.
 
 Similarly, `Chart` resources can enumerate underlying resources and their inputs, thus providing fine-grained diffs and richer previews.
 
 If these are important for your use case, then the `Helm Chart` resource is preferred.
 
+#### Resource Ordering and Readiness
+
+The order in which Kubnernetes resources are applied by Pulumi is based on the dependency
```
Comment (ringods) [inline themes/default/content/registry/packages/kubernetes/how-to-guides/choosing-the-right-helm-resource-for-your-use-case.md:88]: @EronWright typo: `Kubnernetes`

**Review by ringods — CHANGES_REQUESTED** [review]

**Review by ringods — APPROVED** [review]

**Comment by EronWright** [comment]
Update: pulumi-kubernetes v4.12 has been released ([link](https://github.com/pulumi/pulumi-kubernetes/releases/tag/v4.12.0)) including Chart v4 resource, which the revised article refers to.  But we need to hold on this a bit longer, until the api docs ([here](https://www.pulumi.com/registry/packages/kubernetes/api-docs/helm/)) are refreshed.

File: themes/default/content/registry/packages/kubernetes/how-to-guides/choosing-the-right-helm-resource-for-your-use-case.md [hunk 2]
```diff
@@ -5,21 +5,21 @@ aliases: ["/docs/reference/tutorials/kubernetes/chart-vs-release/"]
...
 
 In September 2021 we announced the **public preview** of a new [`Helm Release`](/registry/packages/kubernetes/api-docs/helm/v3/release) which adds an additional option to the mix for Pulumi's Kubernetes users. As of [v3.15.0](https://github.com/pulumi/pulumi-kubernetes/releases/tag/v3.15.0) of the Pulumi Kubernetes SDK and Provider, this resource is now **Generally Available**.
 
-Existing users of the [`Helm V3 Chart`](/registry/packages/kubernetes/api-docs/helm/v3/chart) can continue to use that resource. However, if you are deploying Helm Charts through Pulumi for new use cases, you have a new option to consider. This guide should help you choose the best option for your use case.
+Existing users of the [`Helm Chart`](/registry/packages/kubernetes/api-docs/helm/v4/chart) can continue to use that resource. However, if you are deploying Helm Charts through Pulumi for new use cases, you have a new option to consider. This guide should help you choose the best option for your use case.
```
Comment (pierskarsenbarg) [inline themes/default/content/registry/packages/kubernetes/how-to-guides/choosing-the-right-helm-resource-for-your-use-case.md:12]: Shouldn’t this still be v3 as existing users won’t be using v4 yet?

Comment (EronWright) [reply themes/default/content/registry/packages/kubernetes/how-to-guides/choosing-the-right-helm-resource-for-your-use-case.md:12]: I don't think so, it is saying "use the Chart resource (any version) if you like but also consider using the Release resource".

**Pull request merged**